### PR TITLE
NIFI-10604 - Added idle connection timeout property to ListenTCP. Tes…

### PR DESCRIPTION
**Pull request by thenatog**
…ted using nc utility.

# Summary

[NIFI-10604](https://issues.apache.org/jira/browse/NIFI-10604)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-10604`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-10604`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [x] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by thenatog** [comment]
I tested this idle connection timeout using the ListenTCP processor configured with a listen port of 10054, and the idle timeout set to 5 seconds. I ran the command "nc 127.0.0.1 10054", which opens a TCP connection to 10054, accepting text input from the command line. You can write a message as input to the command, hit return, and wait 5 seconds and the command will terminate, as the connection has been closed by ListenTCP. I believe also waiting 5 seconds and sending no message will also have the connection close.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenTCP.java [hunk 1]
```diff
@@ -133,6 +135,15 @@ public class ListenTCP extends AbstractProcessor {
             .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
             .build();
 
+    protected static final PropertyDescriptor IDLE_CONNECTION_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("idle-timeout")
+            .displayName("Idle Connection Timeout")
+            .description("The amount of time a client's connection will remain open if no data is received. The default of 0 seconds will leave connections open until they are closed by the client.")
+            .required(false)
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenTCP.java:142]: With the default value matching current behavior, it looks like `required` should be `true`, which might also warrant adjusting the order of the property.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the improvement @thenatog! The property works as expected when testing with various values. I just noted one minor adjustment to make the new property required given the available default value.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the update @thenatog, looks good! +1 merging

**Pull request closed without merging**